it/src/main/java/com/google/cloud/teleport/it/datastream/DatastreamResourceManager.java

```diff
@@ -0,0 +1,392 @@
+
+        LOG.info("Updating {}'s state to {} in project {}.", streamId, state.name(), projectId);
+
+        try {
+            Stream.Builder streamBuilder = Stream.newBuilder()
+                    .setName(StreamName.format(projectId, location, streamId))
+                    .setState(state);
+
+            FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+            UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                    .setStream(streamBuilder)
+                    .setUpdateMask(fieldMaskBuilder)
+                    .build();
+
+            Stream reference = datastreamClient.updateStreamAsync(request).get();
+
+            LOG.info("Successfully updated {}'s state to {} in project {}.", streamId, state.name(), projectId);
+            return reference;
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatastreamResourceManagerException("Failed to update stream. ", e);
+        }
+    }
+
+    public synchronized Stream startStream(String streamId) {
+        LOG.info("Starting Stream {} in project {}.", streamId, projectId);
+        return updateStreamState(streamId, Stream.State.RUNNING);
+    }
+
+    public synchronized Stream pauseStream(String streamId) {
+        LOG.info("Pausing Stream {} in project {}.", streamId, projectId);
+        return updateStreamState(streamId, Stream.State.PAUSED);
+    }
+
+
+    public synchronized void cleanupAll() {
+        LOG.info("Cleaning up Datastream resource manager.");
+
+        try {
+            for (String stream : createdStreamIds) {
+                datastreamClient.deleteStreamAsync(DeleteStreamRequest
+                        .newBuilder()
+                        .setName(StreamName.format(projectId, location, stream))
+                        .build()).get();
+            }
+            LOG.info("Successfully deleted stream(s). ");
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatastreamResourceManagerException("Failed to delete stream(s). ", e);
+        }
+
+        try {
+            for (String connectionProfile : createdConnectionProfileIds) {
+                datastreamClient.deleteConnectionProfileAsync(DeleteConnectionProfileRequest
+                        .newBuilder().setName(ConnectionProfileName.format(projectId, location, connectionProfile))
+                        .build()).get();
+            }
+            LOG.info("Successfully deleted connection profile(s). ");
+        } catch (InterruptedException | ExecutionException e) {
+            throw new DatastreamResourceManagerException("Failed to delete connection profile(s). ", e);
+        }
+
+        try {
+            datastreamClient.close();
+        } catch (Exception e) {
+            throw new DatastreamResourceManagerException("Failed to close datastream client. ", e);
+        }
+
```

Comment:
Throw the error at the end.
```suggestion
if (producedError) {
    throw new DatastreamResourceManagerException("Failed to delete resources. Check above for errors.");
}
```